### DISCUSSION TITLE
Name Demangling for C++

### DISCUSSION THREAD

**Discussion by jovere** [danielpalme/ReportGenerator#647] [Q&A] [answered]
Hi,

I'm using ReportGenerator in Azure Devops for a C++ project with LCov/GCov. Importing the LCov `info` file works great. However, the list of functions on the right side of a file is still showing the C++ mangled names, such as `_ZN3RTT8SettingsC2Ev`. Is there a way to feed the data through a demangler so it shows human readable function calls?

Thanks,
Jeremy

**Comment by danielpalme** [comment]
Hi Jeremy,
do you know how to demangle those method names can be done? Is there a C# algorithm which can be used?
I just found http://demangler.com/, it seems to work quite well. But it's an online tool.

**Comment by jovere** [comment, marked as answer]
Hi @danielpalme,
I didn't realize it, but it appears that just running the `.info` file through `c++filt` (a command line tool) after its creation is enough for the system to work. I had incorrectly thought it would break the linking between the code listing and the name links on the right side.
Apologies for bothering you.

**Reply by danielpalme** [reply]
Great that you have found a solution!
